src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -240,25 +374,92 @@ inline DocumentStatisticsAnnotationData::DocumentStatisticsAnnotationData(Freque
 
 // ----------------------------------------------------------------------
 // |
-// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |  DocumentStatisticsEstimator
 // |
 // ----------------------------------------------------------------------
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(void) :
-    _totalNumDocuments(0) {
+//Would it be better to place default values during pipeline chaining(countvectorizer and tfidfvectorizer)? if so this ctor will show no default values
+template <size_t MaxNumTrainingItemsV>
+DocumentStatisticsEstimator<MaxNumTrainingItemsV>::DocumentStatisticsEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex,
+                                                                               StringDecorator decorator, nonstd::optional<IndexMap> vocabulary, 
+                                                                               nonstd::optional<std::uint32_t> maxFeatures, std::float_t minDf, std::float_t maxDf) :
+    BaseType(
+        std::move(pAllColumnAnnotations),
+        std::move(colIndex),
+        true,
+        std::move(decorator),
+        std::move(vocabulary),
+        std::move(maxFeatures),
+        std::move(minDf),
+        std::move(maxDf)
+    ) {
 }
 
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator) :
+// ----------------------------------------------------------------------
+// |
+// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |
+// ----------------------------------------------------------------------
+inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator, nonstd::optional<IndexMap> vocabulary, 
+                                                                                           nonstd::optional<std::uint32_t> maxFeatures, std::float_t minDf, 
```

Comment:
I forget how maxFeatures is used. Is the the max vocab size? max num documents? max number of documents to look at? Consider making this a more descriptive name.

Reply:
it represents a TermFrequencyMap will be trimmed to a new map with words that have "top maxFeatures" of values.
e.g. Let's have an TermFrequencyMap of {{a, 3}, {b, 5}, {c, 1}, {d, 2}, {e, 6}}, assuming maxFeatures = 2, the TermFrequencyMap will be trimmed to { {b, 5},{e, 6} }; if maxFeatures = 4, the TermFrequencyMap will be trimmed to {{a, 3}, {b, 5},{d, 2}, {e, 6}}.

I think "TopKTerms" would be a better one